### PR TITLE
Changed DataTable to add disabled

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add `disabled` which takes an array similar to `select` to specify which rows should be disabled from interaction. The two interactions that can be disabled are selection, when using `onSelect`, and row clicking, when using `onClickRow`.

#### Where should the reviewer start?

index.d.ts

#### What testing has been done on this PR?

Added unit test.
Enhanced Clickable and On Select stories to have a disabled row ("Matt").

#### How should this be manually tested?

stories mentioned above

#### Do Jest tests follow these best practices?

new test is consistent with existing tests

#### What are the relevant issues?

#6067 

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Body.js [hunk 1]
```diff
@@ -182,7 +184,9 @@ const Body = forwardRef(
               }
             : undefined
         }
+        // TODO: skip disabled
```
Comment (jcfilben) [inline src/js/components/DataTable/Body.js:187]: Should we create an issue to change the arrow key behavior for disabled options to skip them?

**Review by jcfilben — APPROVED** [review]
Looks good, we will need to update the docs on grommet-site to include the `disabled` prop

**Review by jcfilben — COMMENTED** [review]
When you are using a screenreader there isn't any indication that a row is disabled. When you try to interact with a disabled row/checkbox it make a noise that indicates you were able to interact with the row/checkbox, but nothing will happen.

**Comment by ericsoderberghp** [comment]
> When you are using a screenreader there isn't any indication that a row is disabled. When you try to interact with a disabled row/checkbox it make a noise that indicates you were able to interact with the row/checkbox, but nothing will happen.

Any suggestions on what should happen?

**Comment by jcfilben** [comment]
When you can click on a row I think we need to add `aria-disabled` to StyledDataTableRow for the rows that are disabled. When there is a disabled checkbox we should add `aria-disabled` to the Cell that contains the checkbox. Ideally the screen reader should read 'disabled' or 'dimmed' and then the name of the cell and then when you attempt to interact with it it shouldn't make the selected noise.

**Pull request merged**